pilot/pkg/proxy/envoy/v2/discovery.go

```diff
@@ -179,8 +174,10 @@ func NewDiscoveryServer(
 		edsUpdates:              map[string]struct{}{},
 		proxyUpdates:            map[string]struct{}{},
 		concurrentPushLimit:     make(chan struct{}, features.PushThrottle),
-		updateChannel:           make(chan *updateReq, 10),
-		pushQueue:               NewPushQueue(),
+		updateChannel: NewDebouncedChannel(DebounceAfter, DebounceMax, false, func(acc DebouncedEvent, v DebouncedEvent) DebouncedEvent {
```

Comment:
The function param can be omitted, pass it here makes things compl

Reply:
No, the merge strategy of events should't be hard wired in the implementation. Other users might like to choose a different merge algorithm.